lstchain/hdu/hdu_table.py

```diff
@@ -0,0 +1,350 @@
+import numpy as np
+import os
+from lstchain.reco.utils import camera_to_altaz
+
+import astropy.units as u
+from astropy.table import Table, Column, vstack, QTable
+from astropy.io import fits
+from astropy.time import Time
+
+__all__ = [
+    'create_obs_hdu_index',
+    'create_event_list'
+    ]
+
+
+DEFAULT_HEADER = fits.Header()
+DEFAULT_HEADER["HDUDOC"] = "https://github.com/open-gamma-ray-astro/gamma-astro-data-formats"
+DEFAULT_HEADER["HDUVERS"] = "0.2"
+DEFAULT_HEADER["HDUCLASS"] = "GADF"
+
+#IRF should be separate?
+def create_obs_hdu_index(filename_list, fits_dir):
+    """
+    Create the obs table and hdu table (below some explanation)
+
+    A two-level index file scheme is used in the IACT community to allow an arbitrary folder structures
+    For each directory tree, two files should be present:
+    **obs-index.fits.gz**
+    (defined in http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/obs_index/index.html)
+    **hdu-index.fits.gz**
+    (defined in http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/hdu_index/index.html)
+
+    obs-index contains the single run informations e.g.
+    (OBS_ID, RA_PNT, DEC_PNT, ZEN_PNT, ALT_PNT)
+    while hdu-index contains the informations about the locations of the other HDU (header data units)
+    necessary to the analysis e.g. A_eff, E_disp and so on...
+        http://gamma-astro-data-formats.readthedocs.io/en/latest/
+
+    This function will create the necessary data format, starting from the path that contains the DL3
+    converted fits file.
+
+    Parameters
+    ----------
+    filename_list : list
+        list of filenames
+    fits_dir : str
+        directory containing the fits file
+    """
+
+    # empty lists with the quantities we want
+    hdu_tables = []
+
+    obs_id = []
+    ra_pnt = []
+    dec_pnt = []
+    zen_pnt = []
+    alt_pnt = []
+    az_pnt = []
+    ontime = []
+    livetime = []
+    deadc = []
+    tstart = []
+    tstop = []
+    source = []
+    N_TELS = []
+    TELLIST = []
+
+    # loop through the files
+    for file in filename_list:
+
+        if os.path.exists(str(fits_dir)+"/"+file):
+            try:
+                event_table = Table.read(str(fits_dir)+"/"+file, hdu="EVENTS")
+                gti_table = Table.read(str(fits_dir)+"/"+file, hdu="GTI")
+                pointing_table = Table.read(str(fits_dir)+"/"+file, hdu="POINTING")
+            except Exception:
+                print(f"fits corrupted for file {file}")
+                continue
+        else:
+            print(f"fits {file} doesn't exist")
```

Comment:
This should also raise an error. The user wanted a file to be included, if it is not there, ignoring that silently is not good.

Reply:
Where silently is relative, I know, but I think just printing it is not enough